trestle/core/commands/author/ssp.py

```diff
@@ -0,0 +1,364 @@
+        """Initialize the class."""
+        super().__init__(parser, parent, name, out, err)
+        self._param_dict: Dict[str, str] = None
+        self._md_file: MDWriter = None
+        self._alters: List[prof.Alter] = None
+        self._yaml_header: dict = None
+        self._sections: Dict[str, str] = None
+
+    def _init_arguments(self) -> None:
+        file_help_str = 'Name of the profile file'
+        self.add_argument('-f', '--file', help=file_help_str, required=True, type=str)
+        yaml_help_str = 'Path to the yaml header file'
+        self.add_argument('-yh', '--yaml-header', help=yaml_help_str, required=True, type=str)
+        sections_help_str = 'Comma separated list of section:alias pairs for sections to output'
+        self.add_argument('-s', '--sections', help=sections_help_str, required=False, type=str)
+        task_help_str = 'Name of the generated ssp markdown folder'
+        self.add_argument('-o', '--output', help=task_help_str, required=True, type=str)
+        verbose_help_str = 'Display verbose output'
+        self.add_argument('-v', '--verbose', help=verbose_help_str, required=False, action='count', default=0)
+
+    def _run(self, args: argparse.Namespace) -> int:
+        log.set_log_level_from_args(args)
+        trestle_root = fs.get_trestle_project_root(pathlib.Path.cwd())
+        if not trestle_root:
+            logger.warning(f'Current working directory {pathlib.Path.cwd()} is not with a trestle project.')
+            return 1
+        if not fs.allowed_task_name(args.output):
+            logger.warning(f'{args.output} is not an allowed directory name')
+            return 1
+
+        if args.mode == 'setup':
+            profile: prof.Profile
+            _, _, profile = load_distributed(pathlib.Path(f'profiles/{args.file}/profile.json'))
+            cat_href = profile.imports[0].href
+            cat_name = fs.model_name_from_href_str(cat_href)
+
+            _, _, catalog = load_distributed(pathlib.Path(f'catalogs/{cat_name}/catalog.json'))
+
+            try:
+                logging.debug(f'Loading yaml header file {args.yaml_header}')
+                yaml_header = yaml.load(pathlib.Path(args.yaml_header).open('r'), yaml.FullLoader)
+            except yaml.YAMLError as e:
+                logging.warning(f'YAML error loading yaml header for ssp generation: {e}')
+                return 1
+            markdown_path = trestle_root / args.output
+
+            sections = None
+            if args.sections is not None:
+                section_tuples = args.sections.strip("'").split(',')
+                sections = {}
+                for section in section_tuples:
+                    if ':' in section:
+                        s = section.split(':')
+                        sections[s[0]] = s[1]
+                    else:
+                        sections[section] = section
+
+            return self.generate_ssp(catalog, profile, markdown_path, sections, yaml_header)
+        else:
```

Comment:
I would think this should be an elif not an else.

Reply:
handled